### PR TITLE
Make `TestDefaultTemplateRendering` less failure prone

### PR DESCRIPTION
**Pull request by timothyb89**
The test requires the Teleport server to report a sane public proxy address, however occasionally it takes some time to initialize and returns an invalid value until that point.

This adds a wait loop to the client initialization helper to wait up to 10 seconds for the server to return a sane value, which should improve reliability of the test.

### CODE REVIEW

**Comment by timothyb89** [comment]
Example failure: 
```
Original Error: *trace.BadParameterError proxy  has no usable public address: missing port in address
Stack Trace:
	/workspace/tool/tbot/config/configtemplate_ssh_client.go:155 github.com/gravitational/teleport/tool/tbot/config.(*TemplateSSHClient).Render
	/workspace/tool/tbot/renew.go:527 github.com/gravitational/teleport/tool/tbot.renew
	/workspace/tool/tbot/configtemplate_test.go:103 github.com/gravitational/teleport/tool/tbot.TestDefaultTemplateRendering
	/opt/go/src/testing/testing.go:1259 testing.tRunner
	/opt/go/src/runtime/asm_amd64.s:1581 runtime.goexit
User Message: proxy  has no usable public address: missing port in address] tbot/renew.go:528
```

(from https://console.cloud.google.com/cloud-build/builds/4fc6949e-d169-4d35-9961-ff780920b720?project=ci-account)

I guess I should note, I can't reproduce this locally so this fix is more an educated guess than a guarantee. But, it matches similar behavior I've seen with other tests that spin up a Teleport server and hopefully has a similar fix.

File: tool/tbot/testhelpers/srv.go [hunk 1]
```diff
@@ -148,6 +148,27 @@ func MakeDefaultAuthClient(t *testing.T, fc *config.FileConfig) auth.ClientI {
 	client, err := authclient.Connect(context.Background(), authConfig)
 	require.NoError(t, err)
 
+	// Wait for the server to become available.
+	for i := 1; i <= 10; i++ {
+		ping, err := client.Ping(context.Background())
+		if err != nil {
+			t.Logf("server is not yet available, waiting (attempt %d of 10)...", i)
+			time.Sleep(1 * time.Second)
```
Comment (espadolini) [inline tool/tbot/testhelpers/srv.go:156]: 1 whole second between attempts is quite long, perhaps we could do this every quarter?

File: tool/tbot/testhelpers/srv.go [hunk 2]
```diff
@@ -148,6 +148,27 @@ func MakeDefaultAuthClient(t *testing.T, fc *config.FileConfig) auth.ClientI {
 	client, err := authclient.Connect(context.Background(), authConfig)
 	require.NoError(t, err)
 
+	// Wait for the server to become available.
+	for i := 1; i <= 10; i++ {
```
[the remark this answers is not in the document]
Comment (zmb3) [reply tool/tbot/testhelpers/srv.go:152]: +1

File: tool/tbot/testhelpers/srv.go (hunk 1 above)
Comment (timothyb89) [reply tool/tbot/testhelpers/srv.go:156]: Easy enough, will do!

File: tool/tbot/testhelpers/srv.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (timothyb89) [reply tool/tbot/testhelpers/srv.go:152]: That's way better, thanks!

**Review by r0mant — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Review by espadolini — APPROVED** [review]

**Pull request merged**